downloads/swan-lake-release-notes/swan-lake-2201.1.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,688 @@
+            x.push(error("invalid!")); // Allowed now. Previously, failed at runtime.
+        }
+    }
+    ```
+
+- Fixed a bug that allowed additive expressions with operands of types that are union types of different basic types 
+
+    ```ballerina
+    public function main() {
+        int|float a = 4;
+        int|float b = 4.5;
+
+        int _ = a + b; // Now, this results in a compilation error.
+    }
+    ```
+
+To view bug fixes, see the [GitHub milestone for Swan Lake 2201.1.0](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+label%3AType%2FBug+label%3ATeam%2FCompilerFE+milestone%3A%22Ballerina+2201.1.0%22).
+
+## Runtime updates
+
+### Improvements
+
+#### Support to provide values for configurable variables through TOML in-line tables
+
+The configurable feature is improved to support TOML in-line tables through the TOML syntax.
+The values for configurable variables of types `map` and `record` can now be provided using TOML in-line tables.
+Similarly, the values for configurable variables of types array of `map`, array of `record`, and `table` can now be provided using the TOML array of TOML in-line tables.
+
+For example, if the configurable variables are defined in the following way,
+
+```ballerina
+configurable map<anydata> mapVar = ?;
+configurable Person recordVar = ?;
+configurable table<map<int>> tableVar = ?;
+configurable Person[] recordArrayVar = ?;
+
+```
+
+the values can be provided in the `Config.toml` file as follows.
+
+```toml
+mapVar = {a = "a", b = 2, c = 3.4, d = [1, 2, 3]}
+
+recordVar = {name = "Jane"}
+
+tableVar = [{a = 1, b = 2}, {c = 3}, {d = 4, e = 5, f = 6}]
+
+recordArrayVar = [{name = "Tom"}, {name = "Harry"}]
+
+```
+
+#### Improved configurable variables to support tuple types through TOML syntax
+
+The configurable feature is improved to support variables of tuple types through the TOML syntax.
+
+For example, if the tuple-typed configurable variables are defined in the following way,
+
+```ballerina
+configurable [int, string, float, decimal, byte, boolean] simpleTuple = ?;
+configurable [int[], [string, int], map<anydata>, table<map<string>>] complexTuple = ?;
+configurable [int, string, int...] restTuple = ?;
+```
+
+the values can be provided in the `Config.toml` file as follows.
+
+```toml
+simpleTuple = [278, "string", 2.3, 4.5, 2, true]
+
+complexTuple = [[1, 3, 5, 7, 9], ["apple", 2], {name = "Baz Qux", age = 22}, [{a = "a"}, {b = "b", c = "c"}]]
+
+restTuple = [1, "foo", 2, 3, 4, 5]
+```
+
+#### Improved configurable variables to support union types through CLI arguments
+
+The configurable feature is improved to support variables of union types with simple basic typed members through the CLI arguments.
+
+For example, if the configurable variables are defined in the following way,
+
+```ballerina
+configurable float|int|string unionVar = ?; 
+```
+
+the values can be provided via CLI arguments in the following way.
+
+```bash
+bal run -- -Cval=5.0
```

Comment:
Shouln't we change this as `bal run -- -CunionVar=5.0` to be complied with previous definition? @warunalakshitha ?

Reply:
yes. it should be. My bad.
```suggestion
bal run -- -CunionVar=5.0
```